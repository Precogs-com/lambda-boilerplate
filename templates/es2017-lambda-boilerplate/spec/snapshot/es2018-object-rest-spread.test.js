const { transform } = require('./utils');

it('object spread properties', () => {
    const code = transform('var x = { a: 1, b: 2 }; var y = { a: 3, ...x };');
    expect(code).toMatchSnapshot();
});

it('object rest properties', () => {
    const code = transform(
        'var x = { a: 1, b: 2 }; var { a, ...rest } = x; console.log(rest.b);'
    );
    expect(code).toMatchSnapshot();
});
