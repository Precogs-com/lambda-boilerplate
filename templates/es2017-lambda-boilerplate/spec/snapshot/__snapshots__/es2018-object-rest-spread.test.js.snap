// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`object rest properties 1`] = `
"\\"use strict\\";

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var x = { a: 1, b: 2 };
var a = x.a,
    rest = _objectWithoutProperties(x, [\\"a\\"]);

console.log(rest.b);"
`;

exports[`object spread properties 1`] = `
"\\"use strict\\";

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var x = { a: 1, b: 2 };var y = _extends({ a: 3 }, x);"
`;
